packages/fixie-web/src/voice.ts

```diff
@@ -356,20 +351,15 @@ export class VoiceSession {
     this.onInputChange?.(text, final);
   }
 
-  private handleOutputChange(text: string, final: boolean) {
-    console.log(`[voiceSession] output: ${text}`);
-    this.onOutputChange?.(text, final);
-  }
-
   private handleOutputTranscript(msg: any) {
-    if (msg.delta) {
+    if (msg.text) {
+      this.outputTranscript = msg.text;
+    } else if (msg.delta) {
       this.outputTranscript += msg.delta;
-      this.onOutputTranscript?.(this.outputTranscript);
-      if (msg.final) {
-        this.outputTranscript = '';
-      }
-    } else {
-      this.onOutputTranscript?.(msg.text);
+    }
+    this.onOutputChange?.(this.outputTranscript, msg.final);
```

Comment:
In the case where only `text` is filled in, are we OK with this being undefined?

Reply:
I don't understand your question.  What is `this`?  With the exception of onOutputChange, nothing here should be undefined once the server change lands.

Reply:
Perhaps you're referring to the interim case before the server change lands. In that case, AI.Town doesn't use onOutputChange right now, so it won't make any difference.  (It uses onOutputTranscript, which means it won't show agent transcripts until the server piece lands.)

Reply:
I'm going to start merging things so I can get everything landed today. If there are issues, then I'll fix them later. :)